nvflare/private/fed/app/simulator/simulator_runner.py

```diff
@@ -286,18 +288,30 @@ def split_clients(self, clients: [], gpus: []):
     def create_clients(self):
         # Deploy the FL clients
         self.logger.info("Create the simulate clients.")
+        client_count_lock = threading.Lock()
+        clients_created_waiter = threading.Event()
         for client_name in self.client_names:
-            client, self.client_config, self.deploy_args, self.build_ctx = self.deployer.create_fl_client(
-                client_name, self.args
-            )
-            self.federated_clients.append(client)
-            app_root = os.path.join(self.simulator_root, "app_" + client_name)
-            app_custom_folder = os.path.join(app_root, "custom")
-            sys.path.append(app_custom_folder)
+            threading.Thread(
+                target=self.create_client, args=[client_name, client_count_lock, clients_created_waiter]
+            ).start()
 
+        clients_created_waiter.wait()
```

Comment:
we could either use this event wait mechanism, or we can have a for loop to join all these threads.

Reply:
The threading.Event wait() is efficient to use here. Also the waiting logic is fully self contained within the class.